File: MorySnip/Classes/Layers/Layer_Oval.vb
Public Class Layer_Oval
    Inherits Layer

    Public Sub New()

    End Sub

    Public Sub New(Pen As Pen, Brush As Brush, FirstPoint As Point)
        Me.Pen = Pen
        Me.Brush = Brush
        Me.Start(FirstPoint)
    End Sub

    Public Overrides Sub Paint(g As Graphics)
        Dim r As Rectangle = Bounds

        If Me.Fill Then
            g.FillEllipse(Brush, r)
        End If
        g.DrawEllipse(Pen, r)
    End Sub
End Class

File: MorySnip/Classes/Layers/LayerOval.cs
using System.Drawing;

namespace Morysoft.MorySnip;

public class LayerOval : Layer
{
    public LayerOval()
    {
    }

    public LayerOval(Pen Pen, Brush Brush, Point FirstPoint)
    {
        this.Pen = Pen;
        this.Brush = Brush;
        this.Start(FirstPoint);
    }

    public override void Paint(Graphics g)
    {
        var r = this.Bounds;

        if (this.Fill)
        {
            g.FillEllipse(this.Brush, r);
        }

        g.DrawEllipse(this.Pen, r);
    }
}
